lib/fastlane/actions/taxi.rb

```diff
@@ -0,0 +1,79 @@
+
+module Fastlane
+  module Actions
+    module SharedValues
+    end
+
+    class TaxiAction < Action
+
+      def self.run(params)
+        taxi_cmd = [
+          "ios-deploy",
+          params[:extra],
+          "--bundle",
+          params[:ipa]
+
+        ]
+
+        taxi_cmd << "--no-wifi" if params[:skip_wifi]
+        taxi_cmd << ["--id", params[:device_id]] if params[:device_id]
+        Actions.sh(taxi_cmd.join(" "))
+        Helper.log.info "Deployd #{params[:ipa]} to device!"
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Installs an .ipa file on a connected ios-device via usb or wifi, requires 'ios-deploy`"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :extra,
+                                       short_option: "-X",
+                                       env_name: "FL_TAXI_EXTRA",
+                                       description: "Extra Commandline arguments passed to ios-deploy",
+                                       optional: true,
+                                       is_string: true
+                                      ),
+          FastlaneCore::ConfigItem.new(key: :device_id,
+                                       short_option: "-d",
+                                       env_name: "FL_TAXI_DEVICE_ID",
+                                       description: "id of the device / if not set defaults to first found device",
+                                       optional: true,
+                                       is_string: true
+                                      ),
+          FastlaneCore::ConfigItem.new(key: :skip_wifi,
+                                       short_option: "-w",
+                                       env_name: "FL_TAXI_WIFI",
+                                       description: "Do not search for devices via WiFi",
+                                       optional: true,
+                                       is_string: false
+                                      ),
+          FastlaneCore::ConfigItem.new(key: :ipa,
+                                       short_option: "-i",
+                                       env_name: "FL_TAXI_IPA",
+                                       description: "The IPA file to put on the device",
+                                       optional: true,
+                                       is_string: true,
+                                       default_value: Dir["*.ipa"].first,
+                                       verify_block: proc do |value|
+                                                       raise "Could not find ipa file at path '#{value}'" unless File.exist? value
```

Comment:
Could you move these blocks to the left to be only 2 spaces intended from `verify_block`